src/System.Text.Json/src/System/Text/Json/JsonConstants.cs

```diff
@@ -72,6 +72,11 @@ internal static class JsonConstants
         public const int MaxDateTimeUtcOffsetHours = 14; // The UTC offset portion of a TimeSpan or DateTime can be no more than 14 hours and no less than -14 hours.
         public const int DateTimeNumFractionDigits = 7;  // TimeSpan and DateTime formats allow exactly up to many digits for specifying the fraction after the seconds.
         public const int MaxDateTimeFraction = 9_999_999;  // The largest fraction expressible by TimeSpan and DateTime formats
+        public const int DateTimeParseNumFractionDigits = 15; // The maximum number of fraction digits the Json DateTime parser allows
+        public const int MaximumDateTimeOffsetParseLength = (MaximumFormatDateTimeOffsetLength +
```

Comment:
Do we need one for just DateTime? i.e. `MaximumDateTimeParseLength` based of  `MaximumFormatDateTimeLength`?

Reply:
No because `MaximumFormatDateTimeLength` does not account for a (non-)utc offset which a `DateTime` string could have.
`DateTime` and `DateTimeOffset` have the same valid lengths in our ISO 8601 profile, so we can use the same constants to validate.